docs/operating-scylla/procedures/cluster-management/add-node-to-cluster.rst

```diff
@@ -98,11 +98,18 @@ Procedure
         UN  192.168.1.202  91.11 KB   256     32.9%             125ed9f4-7777-1dbn-mac8-43fddce9123e   B1
         UN  192.168.1.203  124.42 KB  256     32.6%             675ed9f4-6564-6dbd-can8-43fddce952gy   B1
 
-#. When the new node status is Up Normal (UN), use :doc:`nodetool cleanup </operating-scylla/nodetool-commands/cleanup>` cleanup command on all the nodes in the cluster except the new node that has just been added. It will remove keys that no longer belong to the node. Run this command one node at a time. It is possible to postpone this step to low demand hours.
+#. When the new node status is Up Normal (UN), run the :doc:`nodetool cleanup </operating-scylla/nodetool-commands/cleanup>` command on all nodes in the cluster except for the new node that has just been added. Cleanup removes keys that were streamed to the newly added node and are no longer owned by the node.
 
     .. note::
 
-       If you are using Scylla Enterprise 2018.1.5 or earlier or Scylla Open source 2.3 or earlier, **do not** run the ``nodetool cleanup`` command before upgrading to the latest release of your branch. See this `issue <https://github.com/scylladb/scylla/issues/3872>`_ for more information.
+       The cleanup process may consume significant resources from the system. It may be optimized using the following instructions.
+       However, it is important that cleanup completes successfully before any node is decommissioned or removed from the cluster, as there is a risk of resurrecting stale data stored on the nodes that require cleanup.
+
+       Tip 1: Batch the cleanup operation when bootstrapping a number of new nodes.  In this case, running cleanup can be postponed until after the last node was added.
```

Comment:
This would be very relevant in adding a DC proceudre

Reply:
There, if I understand correctly, with NetworkTopologyStrategy, cleanup will be required only if the replication factor on the existing DCs is reduced.  Otherwise, data will be streamed to the new nodes, but token ownership on the existing DCs will remain, so no cleanup would be required.

Reply:
@asias is it the case?

Reply:
Yes, for keyspace with NetworkTopologyStrategy, adding a node in one DC will make nodes in other DC to lose tokens ranges. 

But I think we should not make user to worry about this at all. I.e., run cleanup anyway, if it is not needed, scylla core will just do no op.

Reply:
> But I think we should not make user to worry about this at all. I.e., run cleanup anyway, if it is not needed, scylla core will just do no op.

I wish.  Currently, the sstable keeps a single token range covering all tokens it stores.  This is likely to overlap with non-owned ranges on the node and will cause cleanup of the sstable, even not need.
I previously proposed to store a list of token ranges per sstable, up to one range in each owned range and then we can have a much better heuristic that will indeed skip all sstables if cleanup is run with no actual movement of ranges.

I'll open an issue about it.  It came up enough times already... :)

Reply:
https://github.com/scylladb/scylladb/issues/12215